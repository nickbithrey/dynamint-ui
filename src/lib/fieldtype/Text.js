import React from 'react';
import PropTypes from 'prop-types';
import { TextField } from 'office-ui-fabric-react/lib/TextField';

const Text = ({name, label, value, update, readOnly, ...remainingProps}) => (
	<TextField 
		label={label} 
		defaultValue={value} 
		onBlur={e => update(e, e.target.value)} 
		{...remainingProps}
	/>
);

Text.defaultProps = {
	value: ''
}

Text.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string,
	value: PropTypes.string,
	update: PropTypes.func
};

export default Text;