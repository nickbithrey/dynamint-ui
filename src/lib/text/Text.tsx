import * as React from 'react';
import { Text, ITextProps } from 'office-ui-fabric-react/lib/Text';

export interface IText extends ITextProps {
    children: React.ReactNode;
}

const T = ({children, ...settings}: IText) => (
    <Text {...settings}>
        {children}
    </Text>
);

export default T;