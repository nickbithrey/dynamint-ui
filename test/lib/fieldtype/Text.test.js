import React from 'react';
import { shallow } from 'enzyme';

import Text from '~/lib/fieldtype/Text';
import fieldTypes from '~/lib/fieldtype';

const tag = 'StyledTextFieldBase';

describe('<Text /> rendering', () => {

	let props = {
			label: 'label',
			name: 'name',
			value: 'value',
			readOnly: true,
			update: jest.fn(() => [])
	};
	
	let wrapper;
	
	beforeEach(() => {
		wrapper = shallow( <Text {...props} /> );
	})
	
	it('can render text field', () => {
		expect(wrapper).toMatchSnapshot();
	});
		
	it('calls update prop on blur', () => {
		const tagWrapper = wrapper.find(tag);
		expect(tagWrapper).toHaveLength(1);
		const event = {
			preventDefault: () => [],
			target: { value: 'newVal' }
		};
		expect(props.update.mock.calls).toHaveLength(0);
		tagWrapper.simulate('blur', event);
		expect(props.update.mock.calls).toHaveLength(1);
		expect(props.update.mock.calls[0][0]).toBe(event);
		expect(props.update.mock.calls[0][1]).toBe(event.target.value);
	});
		
	it('has default value of empty string', () => {
		props = {
			name: 'name',
			// value is undefined
			update: jest.fn(() => [])
		};
		wrapper = shallow( <Text {...props} /> );

		const tagWrapper = wrapper.find(tag);
		expect(tagWrapper).toHaveLength(1);
		expect(tagWrapper.prop('defaultValue')).toBe('');
	});
		
	it('passes through all additional props supplied', () => {
		props = {
			name: 'name',
			value: 'value',
			update: jest.fn(() => []),
			addProp1: 'p1',
			addProp2: 'p2',
			addProp3: 'p3'
		};
		wrapper = shallow( <Text {...props} /> );
		
		const tagWrapper = wrapper.find(tag);
		expect(tagWrapper).toHaveLength(1);
		expect(tagWrapper.props().addProp1).toBe('p1');
		expect(tagWrapper.props().addProp2).toBe('p2');
		expect(tagWrapper.props().addProp3).toBe('p3');
	});
		
	it('is exported from index.js', () => {
		expect(fieldTypes.text).toBe(Text);
	});

});