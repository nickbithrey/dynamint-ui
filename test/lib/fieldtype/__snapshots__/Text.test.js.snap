// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> rendering can render text field 1`] = `
<StyledTextFieldBase
  defaultValue="value"
  label="label"
  onBlur={[Function]}
/>
`;
